numpyro/infer/reparam.py

```diff
@@ -288,3 +289,27 @@ def transform_sample(self, latent):
         """
         x_unconstrained = self.transform(latent)
         return self.guide._unpack_and_constrain(x_unconstrained, self.params)
+
+
+class CircularReparam(Reparam):
+    def __call__(self, name, fn, obs):
+        original_fn = fn
```

Comment:
No need for this, you can use `fn` directly in the below logic.

Reply:
Makes sense, I've done this too.